src/hotspot/os/aix/os_aix.cpp

```diff
@@ -1051,7 +1051,7 @@ void *os::dll_load(const char *filename, char *ebuf, int ebuflen) {
   result = dll_load_library(filename, ebuf, ebuflen);
   // If the load fails,we try to reload by changing the extension to .a for .so files only.
   // Shared object in .so format dont have braces, hence they get removed for archives with members.
-  if (result == nullptr && pointer_to_dot != nullptr && strcmp(pointer_to_dot, old_extension) == 0) {
+  if (strstr(ebuf, "No such file or directory") != nullptr && result == nullptr && pointer_to_dot != nullptr && strcmp(pointer_to_dot, old_extension) == 0) {
```

Comment:
I would switch the first condition with the second. First check if dll_load_library was not successful and then (if not successful) test the error code.
`if (result == nullptr && strstr(ebuf, "No such file or directory") != nullptr && .....`